### PR TITLE
Rename common_clang to opencl_clang

### PR DESCRIPTION
**Pull request by hewj03**
Rename COMMON_CLANG as well.

common_clang is old name, so it is better to rename it.

### CODE REVIEW

**Comment by hewj03** [comment]
ping

**Comment by haonanya** [comment]
lgtm, @hewj03 , thank you!
@cdai2 , please review it.

**Review by fzou1 — COMMENTED** [review]
LGTM

File: CMakeLists.txt [hunk 1]
```diff
@@ -101,10 +101,10 @@ else (WIN32)
 endif (WIN32)
 
 # set that name of the main output file as a target name
-if (NOT DEFINED COMMON_CLANG_LIBRARY_NAME)
-    set(COMMON_CLANG_LIBRARY_NAME "opencl-clang")
+if (NOT DEFINED OPENCL_CLANG_LIBRARY_NAME)
```
Comment (cdai2) [inline CMakeLists.txt:104]: As we rename this macro.  Do we need to notify IGC compiler team?  Maybe they use "COMMON_CLANG_LIBRARY_NAME" during their compilation.

Comment (hewj03) [reply CMakeLists.txt:104]: library name in igc build is libopencl-clang.so

Comment (hewj03) [reply CMakeLists.txt:104]: e.g. xmain 20230302_160000 nightly is using: /rdrive/ref/opencl/runtime/linux/oclgpu/agama-ci-devel-570-EL86/libopencl-clang.so.11

**Pull request merged**